lib/IDE/ExprContextAnalysis.cpp

```diff
@@ -412,6 +412,17 @@ static void collectPossibleCalleesByQualifiedLookup(
     baseTy = MetatypeType::get(baseTy);
 
   collectPossibleCalleesByQualifiedLookup(DC, baseTy, name, candidates);
+
+  // Add 'subscript<T>(keyPath: KeyPath<Root, T>) -> T'.
+  if (name.getBaseName() == DeclBaseName::createSubscript() &&
+      (baseTy->getAnyNominal() || baseTy->is<ArchetypeType>())) {
```

Comment:
Hm, I didn't know this works:
```swift
func foo(val: (String, Int)) {
  _ = val[keyPath: \.0]
}
```
This limitation is not needed.

Reply:
As for meta type types, the only possible keypath is `\.self` (e.g. `String.self[keyPath: \.self]`) because keypath cannot refer static members. @benlangmuir Do you think limiting this to non-metatype types here makes sense?

Reply:
I ended up with just adding `|| baseTy->is<TupleType>()` here.

Reply:
I have no strong opinion.  Seems fine to me either way.